In [5]:
import numpy as np
import pandas as pd
import re

In [9]:
print("numpy",np.__version__)
print("pandas",pd.__version__)
print("re",re.__version__)
print("scikit-learn",sklearn.__version__)

numpy 2.2.2
pandas 2.2.3
re 2.2.1
scikit-learn 1.6.1


In [2]:
house_df = pd.read_csv("real_estate_properties.csv")
house_df

Unnamed: 0                                               Name  \
0              0            4 BHK Apartment for Sale in Sugam Morya   
1              1    2 BHK Apartment for Sale in Regent Park Kolkata   
2              2         2 BHK Apartment for Sale in Parvati Garden   
3              3       3 BHK Apartment for Sale in Multicon Estelle   
4              4  2 BHK Apartment for Sale in Golpark Cooperativ...   
...          ...                                                ...   
2995        2995  3 BHK Apartment for Sale in Greenfield City Cl...   
2996        2996          2 BHK Apartment for Sale in Classic Tower   
2997        2997     1 BHK Apartment for Sale in Mukundapur Kolkata   
2998        2998  3 BHK Apartment for Sale in Koyla Vihar Vasund...   
2999        2999      3 BHK Apartment for Sale in Lake Town Kolkata   

           Area      Furnishing Transaction         Status     Price  
0     1850 sqft     Unfurnished      Resale  Ready to Move  ₹2.38 Cr  
1      525 sqft     Unfurnished      Resale  Ready to Move   ₹32 Lac  
2      705 sqft     Unfurnished      Resale  Ready to Move   ₹32 Lac  
3     1612 sqft     Unfurnished      Resale  Ready to Move  ₹2.25 Cr  
4      980 sqft  Semi-Furnished      Resale  Ready to Move   ₹50 Lac  
...         ...             ...         ...            ...       ...  
2995   850 sqft  Semi-Furnished      Resale  Ready to Move   ₹50 Lac  
2996  1120 sqft             NaN      Resale  Ready to Move  ₹1.50 Cr  
2997   272 sqft             NaN      Resale  Ready to Move   ₹15 Lac  
2998  1800 sqft             NaN      Resale  Ready to Move  ₹1.35 Cr  
2999   928 sqft             NaN         NaN  Ready to Move   ₹42 Lac  

[3000 rows x 7 columns]

In [3]:
house_df.drop('Unnamed: 0',axis=1,inplace=True)
house_df.head()

Name       Area  \
0            4 BHK Apartment for Sale in Sugam Morya  1850 sqft   
1    2 BHK Apartment for Sale in Regent Park Kolkata   525 sqft   
2         2 BHK Apartment for Sale in Parvati Garden   705 sqft   
3       3 BHK Apartment for Sale in Multicon Estelle  1612 sqft   
4  2 BHK Apartment for Sale in Golpark Cooperativ...   980 sqft   

       Furnishing Transaction         Status     Price  
0     Unfurnished      Resale  Ready to Move  ₹2.38 Cr  
1     Unfurnished      Resale  Ready to Move   ₹32 Lac  
2     Unfurnished      Resale  Ready to Move   ₹32 Lac  
3     Unfurnished      Resale  Ready to Move  ₹2.25 Cr  
4  Semi-Furnished      Resale  Ready to Move   ₹50 Lac

In [4]:
house_df['Name'].sample(20)

1923         3 BHK Apartment for Sale in Multicon Estelle
708                    3 BHK House for Sale in Haridevpur
509         3 BHK Apartment for Sale in Lake Town Kolkata
2371      2 BHK Apartment for Sale in Regent Park Kolkata
88      3 BHK Apartment for Sale in Koyla Vihar Vasund...
2899                6 BHK House for Sale in Bangur Avenue
1736       1 BHK Apartment for Sale in Mukundapur Kolkata
967                     Shop for Sale in Vip Road Kolkata
1243         3 BHK Apartment for Sale in Baguiati Kolkata
1138    3 BHK Apartment for Sale in Koyla Vihar Vasund...
4       2 BHK Apartment for Sale in Golpark Cooperativ...
267        1 BHK Apartment for Sale in Mukundapur Kolkata
1885            2 BHK Apartment for Sale in Classic Tower
1676       1 BHK Apartment for Sale in Mukundapur Kolkata
228                    3 BHK House for Sale in Haridevpur
1427                   3 BHK House for Sale in Haridevpur
2884    2 BHK Apartment for Sale in Golpark Cooperativ...
1015    3 BHK 

In [5]:
house_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3000 entries, 0 to 2999
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Name         3000 non-null   object
 1   Area         3000 non-null   object
 2   Furnishing   2623 non-null   object
 3   Transaction  2900 non-null   object
 4   Status       3000 non-null   object
 5   Price        3000 non-null   object
dtypes: object(6)
memory usage: 140.8+ KB


In [6]:
house_df.isnull().sum()

Name             0
Area             0
Furnishing     377
Transaction    100
Status           0
Price            0
dtype: int64

In [7]:
house_df['Name'].value_counts()

Name
4 BHK Apartment for Sale in Sugam Morya                            100
2 BHK Apartment for Sale in Regent Park Kolkata                    100
2 BHK Apartment for Sale in Parvati Garden                         100
3 BHK Apartment for Sale in Multicon Estelle                       100
2 BHK Apartment for Sale in Golpark Cooperative Housing Society    100
2 BHK Apartment for Sale in Shyam Vihar Phase 2                    100
Office Space for Sale in Ballygunge Kolkata                        100
3 BHK Apartment for Sale in Urban Heights                          100
3 BHK House for Sale in Thakurpukur Kolkata                        100
2 BHK Apartment for Sale in Rajarhat Kolkata                       100
2 BHK Apartment for Sale in Batanagar Kolkata                      100
2 BHK Apartment for Sale in Parnasree Palli Kolkata                100
2 BHK Builder Floor for Sale in Buroshibtalla                      100
3 BHK Apartment for Sale in Baguiati Kolkata                       100
3

In [8]:
house_df['Area'].value_counts()

Area
750 sqft     200
850 sqft     200
580 sqft     177
1850 sqft    100
1612 sqft    100
705 sqft     100
525 sqft     100
980 sqft     100
268 sqft     100
1241 sqft    100
1836 sqft    100
1140 sqft    100
652 sqft     100
820 sqft     100
1000 sqft    100
720 sqft     100
1817 sqft    100
1100 sqft    100
4400 sqft    100
2142 sqft    100
853 sqft     100
1675 sqft    100
386 sqft     100
1120 sqft    100
272 sqft     100
1800 sqft    100
928 sqft     100
985 sqft      23
Name: count, dtype: int64

In [9]:
house_df['Price'].value_counts()

Price
₹32 Lac     200
₹50 Lac     200
₹85 Lac     200
₹35 Lac     200
₹42 Lac     200
₹2.38 Cr    100
₹45 Lac     100
₹1.25 Cr    100
₹38 Lac     100
₹2.25 Cr    100
₹2.80 Cr    100
₹1.60 Cr    100
₹51 Lac     100
₹59 Lac     100
₹60 Lac     100
₹75 Lac     100
₹84 Lac     100
₹30 Lac     100
₹1.50 Cr    100
₹1.45 Cr    100
₹2.50 Cr    100
₹37 Lac     100
₹1.35 Cr    100
₹15 Lac     100
₹55 Lac      77
₹67 Lac      23
Name: count, dtype: int64

In [10]:
house_df['Status'].value_counts()

Status
Ready to Move    3000
Name: count, dtype: int64

In [11]:
house_df['Furnishing'].value_counts()

Furnishing
Unfurnished       1223
Semi-Furnished    1200
Furnished          200
Name: count, dtype: int64

In [12]:
house_df['Transaction'].value_counts()

Transaction
Resale    2900
Name: count, dtype: int64

# About Database Prolem

1. Area is in object form.
2. price is also in form & value of "lac" and "Cr" both
3. In Name column bhk , house type and location marje.
4. Furnishing  have  377 & Transaction have 100 null value

# Data Cleaning & Pre Processing

Remove -->  <br>'Office Space for Sale'<br>'Showroom for Sale '<br>'Shop for Sale'

In [13]:
house_df = house_df[~house_df['Name'].str.contains(r'^(Office Space|Showroom|Shop)', case=False)]
house_df['Name'].value_counts()

C:\Users\prama\AppData\Local\Temp\ipykernel_1464\4098554351.py:1: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  house_df = house_df[~house_df['Name'].str.contains(r'^(Office Space|Showroom|Shop)', case=False)]


Name
4 BHK Apartment for Sale in Sugam Morya                            100
2 BHK Apartment for Sale in Regent Park Kolkata                    100
2 BHK Apartment for Sale in Parvati Garden                         100
3 BHK Apartment for Sale in Multicon Estelle                       100
2 BHK Apartment for Sale in Golpark Cooperative Housing Society    100
2 BHK Apartment for Sale in Shyam Vihar Phase 2                    100
3 BHK Apartment for Sale in Urban Heights                          100
3 BHK House for Sale in Thakurpukur Kolkata                        100
2 BHK Apartment for Sale in Rajarhat Kolkata                       100
2 BHK Apartment for Sale in Batanagar Kolkata                      100
2 BHK Builder Floor for Sale in Buroshibtalla                      100
2 BHK Apartment for Sale in Parnasree Palli Kolkata                100
3 BHK Apartment for Sale in Baguiati Kolkata                       100
3 BHK Apartment for Sale in Batanagar Kolkata                      100
3

<b>Convert--><br>'3 BHK Apartment for Sale in Shyam Vihar ' ---->['BHK : 3','Apartment','Shyam Vihar']<br>
'4 BHK House for Sale in Bangur Avenue' -----> ['BHK : 4','House','Bangur Avenue']


In [14]:
house_df['BHK'] = house_df['Name'].str.extract(r'(\d+)\s*BHK').astype(int)
house_df['House_type'] = house_df['Name'].str.extract(r'BHK\s+(\w+)\s+for', flags=re.IGNORECASE)
house_df['Location'] = house_df['Name'].str.extract(r'in\s+(.+)', flags=re.IGNORECASE)
house_df.head()

C:\Users\prama\AppData\Local\Temp\ipykernel_1464\1172536625.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  house_df['BHK'] = house_df['Name'].str.extract(r'(\d+)\s*BHK').astype(int)
C:\Users\prama\AppData\Local\Temp\ipykernel_1464\1172536625.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  house_df['House_type'] = house_df['Name'].str.extract(r'BHK\s+(\w+)\s+for', flags=re.IGNORECASE)
C:\Users\prama\AppData\Local\Temp\ipykernel_1464\1172536625.py:3: SettingWithCopyWarning: 
A value is trying to be s

Name       Area  \
0            4 BHK Apartment for Sale in Sugam Morya  1850 sqft   
1    2 BHK Apartment for Sale in Regent Park Kolkata   525 sqft   
2         2 BHK Apartment for Sale in Parvati Garden   705 sqft   
3       3 BHK Apartment for Sale in Multicon Estelle  1612 sqft   
4  2 BHK Apartment for Sale in Golpark Cooperativ...   980 sqft   

       Furnishing Transaction         Status     Price  BHK House_type  \
0     Unfurnished      Resale  Ready to Move  ₹2.38 Cr    4  Apartment   
1     Unfurnished      Resale  Ready to Move   ₹32 Lac    2  Apartment   
2     Unfurnished      Resale  Ready to Move   ₹32 Lac    2  Apartment   
3     Unfurnished      Resale  Ready to Move  ₹2.25 Cr    3  Apartment   
4  Semi-Furnished      Resale  Ready to Move   ₹50 Lac    2  Apartment   

                              Location  
0                          Sugam Morya  
1                  Regent Park Kolkata  
2                       Parvati Garden  
3                     Multicon Estelle  
4  Golpark Cooperative Housing Society

<b>Area :  '1850 sqft'	(object) ---> 1850 (int)<br>

In [15]:
house_df['Area(sqft)'] = house_df['Area'].apply(lambda x: int(x.split()[0]))
house_df.head()

C:\Users\prama\AppData\Local\Temp\ipykernel_1464\3717934694.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  house_df['Area(sqft)'] = house_df['Area'].apply(lambda x: int(x.split()[0]))


Name       Area  \
0            4 BHK Apartment for Sale in Sugam Morya  1850 sqft   
1    2 BHK Apartment for Sale in Regent Park Kolkata   525 sqft   
2         2 BHK Apartment for Sale in Parvati Garden   705 sqft   
3       3 BHK Apartment for Sale in Multicon Estelle  1612 sqft   
4  2 BHK Apartment for Sale in Golpark Cooperativ...   980 sqft   

       Furnishing Transaction         Status     Price  BHK House_type  \
0     Unfurnished      Resale  Ready to Move  ₹2.38 Cr    4  Apartment   
1     Unfurnished      Resale  Ready to Move   ₹32 Lac    2  Apartment   
2     Unfurnished      Resale  Ready to Move   ₹32 Lac    2  Apartment   
3     Unfurnished      Resale  Ready to Move  ₹2.25 Cr    3  Apartment   
4  Semi-Furnished      Resale  Ready to Move   ₹50 Lac    2  Apartment   

                              Location  Area(sqft)  
0                          Sugam Morya        1850  
1                  Regent Park Kolkata         525  
2                       Parvati Garden         705  
3                     Multicon Estelle        1612  
4  Golpark Cooperative Housing Society         980

<b>Price : <br> '₹2.38 Cr'	(object) ---> 2.38 (int) in cr <br>
 '₹32 Lac'	(object) ---> 0.32 (int) in cr <br>

In [16]:
house_df['price(cr)'] = house_df['Price'].apply(lambda x: round(float(x.strip('₹').split()[0]) / 100, 2) if 'Lac' in x else float(x.strip('₹').split()[0]))
house_df.head()

C:\Users\prama\AppData\Local\Temp\ipykernel_1464\3216666.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  house_df['price(cr)'] = house_df['Price'].apply(lambda x: round(float(x.strip('₹').split()[0]) / 100, 2) if 'Lac' in x else float(x.strip('₹').split()[0]))


Name       Area  \
0            4 BHK Apartment for Sale in Sugam Morya  1850 sqft   
1    2 BHK Apartment for Sale in Regent Park Kolkata   525 sqft   
2         2 BHK Apartment for Sale in Parvati Garden   705 sqft   
3       3 BHK Apartment for Sale in Multicon Estelle  1612 sqft   
4  2 BHK Apartment for Sale in Golpark Cooperativ...   980 sqft   

       Furnishing Transaction         Status     Price  BHK House_type  \
0     Unfurnished      Resale  Ready to Move  ₹2.38 Cr    4  Apartment   
1     Unfurnished      Resale  Ready to Move   ₹32 Lac    2  Apartment   
2     Unfurnished      Resale  Ready to Move   ₹32 Lac    2  Apartment   
3     Unfurnished      Resale  Ready to Move  ₹2.25 Cr    3  Apartment   
4  Semi-Furnished      Resale  Ready to Move   ₹50 Lac    2  Apartment   

                              Location  Area(sqft)  price(cr)  
0                          Sugam Morya        1850       2.38  
1                  Regent Park Kolkata         525       0.32  
2                       Parvati Garden         705       0.32  
3                     Multicon Estelle        1612       2.25  
4  Golpark Cooperative Housing Society         980       0.50

<b>Drop unnecessary column

In [17]:
house_df = house_df.drop(columns=['Price','Furnishing','Transaction','Name','Area','Status'])
house_df.head()

BHK House_type                             Location  Area(sqft)  price(cr)
0    4  Apartment                          Sugam Morya        1850       2.38
1    2  Apartment                  Regent Park Kolkata         525       0.32
2    2  Apartment                       Parvati Garden         705       0.32
3    3  Apartment                     Multicon Estelle        1612       2.25
4    2  Apartment  Golpark Cooperative Housing Society         980       0.50

In [18]:
house_df['Location'].value_counts()

Location
Batanagar Kolkata                      200
Sugam Morya                            100
Regent Park Kolkata                    100
Parvati Garden                         100
Golpark Cooperative Housing Society    100
Multicon Estelle                       100
Shyam Vihar Phase 2                    100
Urban Heights                          100
Thakurpukur Kolkata                    100
Rajarhat Kolkata                       100
Buroshibtalla                          100
Parnasree Palli Kolkata                100
Baguiati Kolkata                       100
Behala Kolkata                         100
Behala Chowrasta Kolkata               100
Haridevpur                             100
Motilal Gupta Road Kolkata             100
Bangur Avenue                          100
Ballygunge Kolkata                     100
Golf Towers                            100
Greenfield City Classic Premium        100
Classic Tower                          100
Mukundapur Kolkata                     100
Ko

In [19]:
house_df.isnull().sum()

BHK             0
House_type    100
Location        0
Area(sqft)      0
price(cr)       0
dtype: int64

In [20]:
house_df['House_type'] = house_df['House_type'].fillna('Apartment')

In [21]:
house_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2623 entries, 0 to 2999
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   BHK         2623 non-null   int64  
 1   House_type  2623 non-null   object 
 2   Location    2623 non-null   object 
 3   Area(sqft)  2623 non-null   int64  
 4   price(cr)   2623 non-null   float64
dtypes: float64(1), int64(2), object(2)
memory usage: 123.0+ KB


In [22]:
house_df.describe()

BHK   Area(sqft)    price(cr)
count  2623.000000  2623.000000  2623.000000
mean      2.771254  1201.660313     0.904846
std       1.007741   792.103584     0.696859
min       1.000000   272.000000     0.150000
25%       2.000000   750.000000     0.380000
50%       3.000000   980.000000     0.600000
75%       3.000000  1612.000000     1.350000
max       6.000000  4400.000000     2.800000

In [23]:
house_df.columns

Index(['BHK', 'House_type', 'Location', 'Area(sqft)', 'price(cr)'], dtype='object')

In [24]:
house_df = house_df[['Location','House_type','BHK','Area(sqft)', 'price(cr)']]

# Split the dataset

In [25]:
X = house_df.iloc[:,:-1]
y = house_df.iloc[:,-1]

In [8]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression,Lasso,Ridge
from sklearn.preprocessing import StandardScaler,OneHotEncoder
from sklearn.compose import make_column_transformer
from sklearn.pipeline import make_pipeline
from sklearn.metrics import r2_score

In [27]:
X_train,X_test,y_train,y_test = train_test_split(X,y,test_size=0.2,random_state=2)

In [28]:
X_train.shape,X_test.shape

((2098, 4), (525, 4))

# Train and Test the model

In [29]:
column_trans = make_column_transformer(
    (OneHotEncoder(sparse_output=False), ['Location', 'House_type']),
    remainder='passthrough'
)

In [30]:
scaler = StandardScaler()

<b><h3>Linear Regression

In [31]:
lr = LinearRegression()
pipe_lr = make_pipeline(column_trans,scaler,lr)
pipe_lr.fit(X_train,y_train)

C:\Users\prama\AppData\Roaming\Python\Python313\site-packages\sklearn\compose\_column_transformer.py:1667: FutureWarning: 
The format of the columns of the 'remainder' transformer in ColumnTransformer.transformers_ will change in version 1.7 to match the format of the other transformers.
At the moment the remainder columns are stored as indices (of type int). With the same ColumnTransformer configuration, in the future they will be stored as column names (of type str).
To use the new behavior now and suppress this warning, use ColumnTransformer(force_int_remainder_cols=False).

  warnings.warn(


Pipeline(steps=[('columntransformer',
                 ColumnTransformer(remainder='passthrough',
                                   transformers=[('onehotencoder',
                                                  OneHotEncoder(sparse_output=False),
                                                  ['Location',
                                                   'House_type'])])),
                ('standardscaler', StandardScaler()),
                ('linearregression', LinearRegression())])

In [32]:
y_pred_lr = pipe_lr.predict(X_test)

<b><h3>Lasso Regression

In [33]:
lasso = Lasso()
pipe_ls = make_pipeline(column_trans,scaler,lasso)
pipe_ls.fit(X_train,y_train)

C:\Users\prama\AppData\Roaming\Python\Python313\site-packages\sklearn\compose\_column_transformer.py:1667: FutureWarning: 
The format of the columns of the 'remainder' transformer in ColumnTransformer.transformers_ will change in version 1.7 to match the format of the other transformers.
At the moment the remainder columns are stored as indices (of type int). With the same ColumnTransformer configuration, in the future they will be stored as column names (of type str).
To use the new behavior now and suppress this warning, use ColumnTransformer(force_int_remainder_cols=False).

  warnings.warn(


Pipeline(steps=[('columntransformer',
                 ColumnTransformer(remainder='passthrough',
                                   transformers=[('onehotencoder',
                                                  OneHotEncoder(sparse_output=False),
                                                  ['Location',
                                                   'House_type'])])),
                ('standardscaler', StandardScaler()), ('lasso', Lasso())])

In [34]:
y_pred_lasso = pipe_ls.predict(X_test)

<b><h3>Ridge Regression

In [35]:
ridge = Ridge()
pipe_rd = make_pipeline(column_trans,scaler,ridge)
pipe_rd.fit(X_train,y_train)

C:\Users\prama\AppData\Roaming\Python\Python313\site-packages\sklearn\compose\_column_transformer.py:1667: FutureWarning: 
The format of the columns of the 'remainder' transformer in ColumnTransformer.transformers_ will change in version 1.7 to match the format of the other transformers.
At the moment the remainder columns are stored as indices (of type int). With the same ColumnTransformer configuration, in the future they will be stored as column names (of type str).
To use the new behavior now and suppress this warning, use ColumnTransformer(force_int_remainder_cols=False).

  warnings.warn(


Pipeline(steps=[('columntransformer',
                 ColumnTransformer(remainder='passthrough',
                                   transformers=[('onehotencoder',
                                                  OneHotEncoder(sparse_output=False),
                                                  ['Location',
                                                   'House_type'])])),
                ('standardscaler', StandardScaler()), ('ridge', Ridge())])

In [36]:
y_pred_ridge = pipe_rd.predict(X_test)

<b><h3>R2 score :

In [37]:
print("R2 score of Linear Regression",r2_score(y_test,y_pred_lr))
print("R2 score of Ridge Regression",r2_score(y_test,y_pred_ridge))
print("R2 score of Lasso Regression",r2_score(y_test,y_pred_lasso))

R2 score of Linear Regression 1.0
R2 score of Ridge Regression 0.9999991512900006
R2 score of Lasso Regression -0.0069322006409389125


<b>Linear Regression gave the parfect value so i go with linear Regression

In [114]:
import pickle
pickle.dump(pipe_lr,open('LinearRegressionModel.pkl','wb'))
pickle.dump(house_df,open('HouseData.pkl','wb'))

<b>Try prediction

In [109]:
d = pd.DataFrame([['Thakurpukur Kolkata', 'Apartment', 2, 500]],
                 columns=['Location', 'House_type', 'BHK', 'Area(sqft)'])
d

Location House_type  BHK  Area(sqft)
0  Thakurpukur Kolkata  Apartment    2         500

In [113]:
prediction = pipe_lr.predict(d)
prediction*10000000

array([3157560.54609725])

In [115]:
house_df.columns

Index(['Location', 'House_type', 'BHK', 'Area(sqft)', 'price(cr)'], dtype='object')

In [116]:
input_data = pd.DataFrame([{
        'Location': 'Thakurpukur Kolkata',
        'House_type': 'Apartment',
        'BHK': 2,
        'Area(sqft)': 500
    }])

In [119]:
prediction = pipe_lr.predict(input_data)
prediction

array([0.31575605])